nsxt/resource_nsxt_edge_cluster.go

```diff
@@ -95,6 +103,12 @@ func resourceNsxtEdgeCluster() *schema.Resource {
 					},
 				},
 			},
+			"failure_domain_allocation": {
+				Type:         schema.TypeString,
```

Comment:
If we only have 2 options, why not use `TypeBool`?

Reply:
Actually, we have three options: 1. enable 2. disable 3. unset. In API, there is a list to provide the allocation rules, which could be empty or contain a rule enabling failure domain allocation or a rule disabling failure domain allocation. It's too complicated for users to just enable the failure domain, so we simplify it in our terraform schema here. 

I'm not sure what the difference is between an empty list and a disabling failure domain rule in the actual behavior. The reason I use string here is similar to this discussion https://github.com/vmware/terraform-provider-nsxt/pull/957#issuecomment-1718370583  if we use bool here, then when a user imports an edge cluster without failure domain preference, we will automatically add a disable failure domain allocation rule inside.

Reply:
If we're unsure about platform behavior, lets keep the three options, but rename them in a way that would not cause confusion regarding property type (its not intuitive for terraform user to wrap `true` and `false` values in quotes)

Reply:
Make sense. Done.